### ISSUE TITLE
source code compilation error

### ISSUE THREAD

**Issue by xingwozhonghua126** [viennashe/viennashe-dev#30]
[http://viennashe.sourceforge.net/doc/manual-page-installation.html](http://viennashe.sourceforge.net/doc/manual-page-installation.html)

windows
D8021 Invalid numeric parameter "/Wl,-rpath"

========== "build": 0 succeeded, 28 failed, 1 updated, 8 skipped ==========

linux
```

iennashe-dev/build$ make
[  1%] Building CXX object src/CMakeFiles/shesolvers.dir/solver.cpp.o
In file included from /home/uos/桌面/viennashe-dev/src/solver.cpp:23:
/home/uos/桌面/viennashe-dev/viennashe/forwards.h:52:10: fatal error: viennagrid/forwards.hpp: 没有那个文件或目录
   52 | #include "viennagrid/forwards.hpp"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [src/CMakeFiles/shesolvers.dir/build.make:76：src/CMakeFiles/shesolvers.dir/solver.cpp.o] 错误 1
make[1]: *** [CMakeFiles/Makefile2:1065：src/CMakeFiles/shesolvers.dir/all] 错误 2
make: *** [Makefile:146：all] 错误 2
```

**Comment by karlrupp** [comment]
Please download the release tarball from https://sourceforge.net/projects/viennashe/ . Cloning the developer repository requires additional steps.

**Comment by xingwozhonghua126** [comment]
[uos@uos-vmwarevirtualplatform build]$ make
[  1%] Building CXX object src/CMakeFiles/shesolvers.dir/solver.cpp.o
In file included from /home/uos/下载/ViennaSHE-1.3.0/src/solvers/solve.hpp:25,
                 from /home/uos/下载/ViennaSHE-1.3.0/src/solver.cpp:25:
/home/uos/下载/ViennaSHE-1.3.0/src/solvers/petsc/petsc_solver.hpp:25:10: 致命错误：petscksp.h：没有那个文件或目录
   25 | #include <petscksp.h>
      |          ^~~~~~~~~~~~
编译中断。
make[2]: *** [src/CMakeFiles/shesolvers.dir/build.make:76：src/CMakeFiles/shesolvers.dir/solver.cpp.o] 错误 1
make[1]: *** [CMakeFiles/Makefile2:1020：src/CMakeFiles/shesolvers.dir/all] 错误 2
make: *** [Makefile:146：all] 错误 2
``

use https://sourceforge.net/projects/viennashe/  the same status

**Comment by karlrupp** [comment]
This means that PETSc is not installed. `A working PETSc installation is required.`
